idaes/apps/grid_integration/bidder.py

```diff
@@ -1291,3 +1291,413 @@ def _record_bids(self, bids, date, hour, **kwargs):
+        """
+        No need to update the RT bidding problem here.
+
+        Arguments:
+            None
+
+        Returns:
+            None
+        """
+        pass
+
+    def record_bids(self, bids: dict, model, date: str, hour: int, market):
+        """
+        This function records the bids and the details in the underlying bidding model.
+        The detailed bids of each time step at day-ahead and real-time planning horizon will
+        be recorded at bidder_detail.csv
+
+        Arguments:
+            bids: dictionary,
+                the obtained bids for this date. Keys are time step t, example as following:
+                bids = {t: {gen: {p_cost: float, p_max: float, p_min: float, startup_capacity: float, shutdown_capacity: float}}}
+            model: pyomo model object,
+                our bidding model.
+            date: str,
+                the date we bid into.
+            hour: int,
+                the hour we bid into.
+            market: str,
+                the market we participate.
+
+        Returns:
+            None
+
+        """
+
+        # record bids
+        self._record_bids(bids, date, hour, Market=market)
+
+        return
+
+    def _record_bids(self, bids: dict, date: str, hour: int, **kwargs):
+        """
+        Record the bis of each time period.
+
+        Arguments:
+            bids: dictionary,
+                the obtained bids for this date. Keys are time step t, example as following:
+                bids = {t: {gen: {p_cost: float, p_max: float, p_min: float, startup_capacity: float, shutdown_capacity: float}}}
+            date: str,
+                the date we bid into.
+            hour: int,
+                the hour we bid into.
+
+        Returns:
+            None
+
+        """
+        df_list = []
+        for t in bids:
+            for gen in bids[t]:
+
+                result_dict = {}
+                result_dict["Generator"] = gen
+                result_dict["Date"] = date
+                result_dict["Hour"] = t
+
+                for k, v in kwargs.items():
+                    result_dict[k] = v
+
+                num_bid_pairs = len(bids[t][gen]["p_cost"])
+
+                for idx, (power, cost) in enumerate(bids[t][gen]["p_cost"]):
+                    result_dict[f"Power {idx} [MW]"] = power
+                    result_dict[f"Cost {idx} [$]"] = cost
+
+                result_df = pd.DataFrame.from_dict(result_dict, orient="index")
+                df_list.append(result_df.T)
+
+        # save the result to object property
+        # wait to be written when simulation ends
+        self.bids_result_list.append(pd.concat(df_list))
+
+        return
+
+    def write_results(self, path):
+        """
+        This methods writes the saved operation stats into an csv file.
+
+        Arguments:
+            path: str or Pathlib object,
+                the path to write the results.
+
+        Return:
+            None
+        """
+
+        print("")
```

Comment:
Using a logger is generally preferred to doing `print` statements as this gives the end-user more flexibility in how (and what) things are displayed.

Also, for a new line use `print("\n"}`.

Reply:
https://github.com/IDAES/idaes-pse/blob/main/.github/CODEOWNERS

Reply:
I saw @adowling2 is on the CODEOWNERS roster for the grid_integration codes. I am a PhD student and will graduate in 2 years but I am happy to be responsible for maintaining these codes until I graduate.

Reply:
> Using a logger is generally preferred to doing `print` statements as this gives the end-user more flexibility in how (and what) things are displayed.
> 
> Also, for a new line use `print("\n"}`.

Done.

Reply:
I see that you changed the new line, but what about using a logger?

Reply:
I am not very familiar with the logger, so I need to look into that.

Reply:
Hi Andrew, I used logger to replace print and it works. Thank you for letting me know this.